forte/utils/utils_processor.py

```diff
@@ -80,3 +83,38 @@ def parse_allennlp_srl_tags(tags: str) -> \
             end = i
 
     return pred_span, arguments
+
+
+def record_types_and_attributes_check(expectation: Dict[str, List[str]],
+                                      input_pack: PackType):
+    r"""Check if any types or attributes in expectation dictionary doesn't
+    match with input_pack.record. If not, an error of
+    :class:`~forte.common.exception.ExpectedRecordNotFound` will be raised.
+
+    Args:
+        expectation: Dictionary of types and their attributes required for
+            the current processor/evaluator.
+        input_pack: The input pack.
+
+    Returns:
+
+    """
+    # pylint: disable=protected-access
+    if expectation is not None:
+        # check if expected types are in input pack.
+        for expected_t in expectation:
+            if expected_t not in input_pack._meta.record.keys():
+                raise ExpectedRecordNotFound(
+                    f"The record type {expected_t} is not found in "
+                    f"meta of the prediction datapack.")
+            else:
+                expected_value = expectation.get(expected_t)
+                if expected_value is not None:
+                    for expected_t_v in expected_value:
+                        if expected_t_v not in input_pack._meta.record \
```

Comment:
If we are checking for `contains` (i.e x in y), should we use `set()` instead of `list()`?